at-date-lib/src/main/kotlin/com/amerharb/atdate/Encode.kt

```diff
@@ -0,0 +1,366 @@
+package com.amerharb.atdate
+
+import kotlin.math.absoluteValue
+
+fun encode(input: String): AtDate {
+	return if (input.startsWith("@P")) {
+		encodePeriod(input)
+	} else {
+		encodeMoment(input)
+	}
+}
+
+fun encodeMoment(input: String): Moment {
+	// takes input like "@2019-01-01T00:00:00Z {d:1 t:5 z:1 a:s l:0-0}@"
+	// takes input like "@1979-11-14 {d:1 t:5 z:1 a:s l:0-0}@"
+	// and returns an Moment object
+	val ad = input.trim().substringAfter("@").substringBefore("@").trim()
```

Comment:
Well this PR is just for format and linting, so I would do another refactor PR later. so this out of scope for now
btw this file is not new it is just renamed but because the changes effect most of the lines (cuz of indentation change) so git think it is a new file